fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3969,14 +3990,13 @@ public ChangedTransactionDetail close(final JsonCommand command, final LoanLifec
             final Money totalLoanOverpayment = calculateTotalOverpayment();
             if (totalLoanOverpayment.isGreaterThanZero() && getInArrearsTolerance().isGreaterThanOrEqualTo(totalLoanOverpayment)) {
                 // TODO - KW - technically should set somewhere that this loan
-                // has
-                // 'overpaid' amount
+                // has 'overpaid' amount
+                this.closedOnDate = closureDate;
```

Comment:
Why to move this date?

Reply:
I needed to move the set of the `closedOnDate` value before the Life cycle machine method call because when the event listener was called It was in null at that moment